client/landing/subscriptions/components/add-sites-form/add-sites-form.tsx

```diff
@@ -45,7 +49,16 @@ const AddSitesForm = ( {
 	const { mutate: subscribe, isPending: subscribing } =
 		SubscriptionManager.useSiteSubscribeMutation();
 
-	const validateInputValue = ( url: string, showError = false ) => {
+	// Triggers the onBlur event when the component mounts to validate the initial value.
+	useEffect( () => ( searchTerm ? onTextFieldChange( searchTerm ) : undefined ), [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	// Update url query when search term changes.
+	useEffect(
+		() => setUrlQuery( SEARCH_QUERY_PARAM, searchTerm, pathname ),
+		[ searchTerm, pathname ]
+	);
```

Comment:
Im just wondering if this needs a useEffect, or if we could call `setUrlQuery` in `onTextFieldChange` ?

Reply:
Using it onTextFieldChange now [7b566f0](https://github.com/Automattic/wp-calypso/pull/101625/commits/7b566f0ed7e8f6deadded60a135b135520621e25).

Thanks for the good suggestion.